corehq/apps/app_manager/xform.py

```diff
@@ -948,6 +948,16 @@ def get_questions(self, langs, include_triggers=False,
                     "type": "DataBindOnly",
                     "calculate": bind.attrib.get('calculate') if hasattr(bind, 'attrib') else None,
                 }
+
+                # Include meta information about the stock entry
+                if data_node.tag_name == 'entry':
+                    parent = next(data_node.xml.iterancestors())
+                    if parent and WrappedNode(parent).tag_name in STOCK_QUESTION_TAG_NAMES:
+                        question.update({
+                            "stock_entry_attributes": data_node.xml.attrib,
+                            "stock_type_attributes": parent.attrib,
+                        })
```

Comment:
This is actually something I thought of when I was freaking out over all the edge cases not including the question id in the form xml path would cause. when I tried to do what you suggested it gave me a generic build error. Though I do agree testing for xmlns is more correct and my way isn't foolproof. Gonna update